falguard.py

```diff
@@ -69,13 +69,28 @@ def __init__(self, spec_path):
     def __call__(self, request, response, resource, params):
         self.process_resource(request, response, resource, params)
 
+    def _build_http_not_allowed_error(self, uri):
+        # Error 405 requires we provide the list of allowed methods
+        # for this URI. If None is found, then we produce a 404 error.
+        allowed_methods = [method for method in falcon.HTTP_METHODS
+                           if self._spec.get_op_for_request(method, uri)]
+
+        return (falcon.HTTPMethodNotAllowed(allowed_methods) if allowed_methods
+                else falcon.HTTPNotFound)
+
     def process_resource(self, request, response, resource, params):
         bravado_request = _BravadoRequest(request, params)
 
+        uri = request.uri_template
         operation = self._spec.get_op_for_request(
-            request.method.upper(), request.uri_template)
+            request.method.upper(), uri)
         if not operation:
-            raise falcon.HTTPNotFound
+            if uri:
+                # The URI exists but the method is wrong.
+                # So we are going to reply with an error 405.
+                raise self._build_http_not_allowed_error(uri)
+            else:
+                raise falcon.HTTPNotFound
```

Comment:
As mentioned in one of the previous comments, you can move this logic right after line 84. What I didn't mention is that this path is not covered by tests. I will need to update Travis to run `py.test ... --cov falguard --cov-fail-under 100`